### PR TITLE
Raise FileNotFoundError when a literal JSON string that looks like a json filename is passed

### LINKED ISSUE #13026: cudf.read_json returns ValueError when given file is not found
**Describe the bug**
The `cudf.read_json` reports an incorrect error when given an invalid file path.

**Steps/Code to reproduce bug**
```
>>> import cudf
>>> df = cudf.read_json("nosuchfile.json")
/conda/envs/rapids/lib/python3.10/site-packages/cudf-23.6.0-py3.10-linux-x86_64.egg/cudf/io/json.py:121: UserWarning: Using CPU via Pandas to read JSON dataset, this may be GPU accelerated in the future
  warnings.warn(
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/conda/envs/rapids/lib/python3.10/site-packages/cudf-23.6.0-py3.10-linux-x86_64.egg/cudf/io/json.py", line 143, in read_json
    pd_value = pd.read_json(
  File "/conda/envs/rapids/lib/python3.10/site-packages/pandas/util/_decorators.py", line 211, in wrapper
    return func(*args, **kwargs)
  File "/conda/envs/rapids/lib/python3.10/site-packages/pandas/util/_decorators.py", line 331, in wrapper
    return func(*args, **kwargs)
  File "/conda/envs/rapids/lib/python3.10/site-packages/pandas/io/json/_json.py", line 757, in read_json
    return json_reader.read()
  File "/conda/envs/rapids/lib/python3.10/site-packages/pandas/io/json/_json.py", line 915, in read
    obj = self._get_object_parser(self.data)
  File "/conda/envs/rapids/lib/python3.10/site-packages/pandas/io/json/_json.py", line 937, in _get_object_parser
    obj = FrameParser(json, **kwargs).parse()
  File "/conda/envs/rapids/lib/python3.10/site-packages/pandas/io/json/_json.py", line 1064, in parse
    self._parse_no_numpy()
  File "/conda/envs/rapids/lib/python3.10/site-packages/pandas/io/json/_json.py", line 1321, in _parse_no_numpy
    loads(json, precise_float=self.precise_float), dtype=None
ValueError: Unexpected character found when decoding 'null'
```
Using `engine="cudf"` gives the following result
```
>>> df = cudf.read_json("nosuchfile.json", engine="cudf")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/conda/envs/rapids/lib/python3.10/site-packages/cudf-23.6.0-py3.10-linux-x86_64.egg/cudf/io/json.py", line 111, in read_json
    df = libjson.read_json(
  File "json.pyx", line 50, in cudf._lib.json.read_json
  File "json.pyx", line 138, in cudf._lib.json.read_json
RuntimeError: CUDF failure at: /cudf/cpp/src/io/json/json_column.cu:958: Input needs to be an array of arrays or an array of (nested) objects
```

I spent way too much time trying to debug my json file format until I realized I had a typo in the path name.

**Expected behavior**
The other `cudf.read_*` functions raise `FileNotFoundError` if the file cannot be found

### PR DESCRIPTION
**Pull request by lithomas1**
## Description

- closes #13026 

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by lithomas1** [comment]
Talked on slack with Vyas, we concluded that it's probably too early to rip this stuff out.

We probably can be a bit smarter for cases where a JSON string looks like a json filename, though.

the relevant logic for pandas is https://github.com/pandas-dev/pandas/pull/46718/files#diff-472ce5fe087e67387942e1e1c409a5bc58dde9eb8a2db6877f1a45ae4974f694R724-R728

File: python/cudf/cudf/utils/ioutils.py [hunk 1]
```diff
@@ -1718,10 +1718,32 @@ def get_reader_filepath_or_buffer(
...
             if len(paths):
                 if fs.exists(paths[0]):
                     path_or_data = paths if len(paths) > 1 else paths[0]
-                elif not allow_raw_text_input:
+
+                # raise FileNotFound if path looks like json
+                # following pandas
+                # see
+                # https://github.com/pandas-dev/pandas/pull/46718/files#diff-472ce5fe087e67387942e1e1c409a5bc58dde9eb8a2db6877f1a45ae4974f694R724-R729
+                elif not allow_raw_text_input or paths[0].lower().endswith(
```
[the remark this answers is not in the document]
Comment (lithomas1) [reply python/cudf/cudf/utils/ioutils.py:1744]: Probably not.

If this heuristic is tripped, we can probably assume the user is passing in a filename, and not a raw JSON string (so the warning is not relevant).

[the remark this answers is not in the document]
Comment (wence-) [reply python/cudf/cudf/utils/ioutils.py:1744]: Ah, I think I was confused, I was worried that we weren't seeing the deprecation warning on line ~1710 if passing a literal string, but the high level caller does produce it. So nothing to do here.

**Review by vyasr — APPROVED** [review]

**Comment by vyasr** [comment]
Looks great to me!

**Pull request merged**